src/lib/support/logging/CHIPLogging.cpp

```diff
@@ -199,6 +199,115 @@ DLL_EXPORT void SetLogFilter(uint8_t category)
 }
 #endif // CHIP_LOG_FILTERING
 
+/**
+ * This static method outputs a line of the memory dump.
+ *
+ * @param[in]  module    The LogModule generating the log message.
+ * @param[in]  category  The LogCategory for filtering log level.
+ * @param[in]  aBuf      A pointer to the buffer.
+ * @param[in]  aLength   Number of bytes in the buffer.
+ *
+ */
+static void DumpLine(uint8_t module, uint8_t category, const void * aBuf, const size_t aLength)
+{
+    char buf[80];
+    char * cur = buf;
+
+    snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), "|");
+    cur += strlen(cur);
+
+    for (size_t i = 0; i < 16; i++)
+    {
+        if (i < aLength)
+        {
+            snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), " %02X", ((uint8_t *) (aBuf))[i]);
+            cur += strlen(cur);
+        }
```

Comment:
All of this code could use `StringBuilder` from src/lib/support/StringBuilder.h, which is both safer and easier to read.

For the hex conversion of current byte, you can do:

```
    char hex_string[3];
    BytesToUppercaseHexString(&buf[i], 1, &hex_string[0], sizeof(hex_string));
    writer.Put(hex_string);
```